src/router.ts

```diff
@@ -2,48 +2,70 @@ import Router from '@koa/router';
 
 import * as k8s from './k8s';
 import { taskStore } from './server';
+import { TaskInfo } from './types';
 import { NAMESPACE } from './constants';
 
 const router = new Router();              //Instantiate the router
 
 // Default empty route
-router.get('/', (ctx, _next) => {
-  ctx.body = taskStore;
+router.get('/task', (ctx, _next) => {
+  if (!ctx.querystring) {
+    ctx.body = taskStore;
+    return;
+  }
+  const { projectName, projectLink,  }  = ctx.query;
+
+  const allTasks: TaskInfo[] = Object.values(taskStore.registeredTasks);
+
+  const filterByName = projectName ? allTasks.filter(t => t.projectName === projectName) : allTasks;
+
+  const queryResult = projectLink ? filterByName.filter(t => t.projectLink === projectLink) : filterByName;
+
+  ctx.body = queryResult;
+
 });
 
 // check Tasks
 router.get('/task/:id', (ctx, _next) => {
   const taskID = ctx.params.id;
 
-  const data = {
-    taskID,
-    taskMeta: taskStore.registeredTasks[taskID],
-    k8sJob: taskStore.k8sJobs[taskID]
+  const data = taskStore.registeredTasks[taskID]
+  if (!data) {
+    ctx.throw(400, 'Invalid task id: ', taskID);
+    return;
   }
 
   ctx.body = data;
 });
 
-function validateTaskRequest(requestBody: Record<string, any>): string | undefined {
-  const { projectLink } = requestBody;
 
-  return projectLink ?? projectLink;
+interface ProjectInfo {
```

Comment:
If you may use this interface in multiple places, you can move this to `src/types.ts`